build.sh

```diff
@@ -177,77 +180,116 @@ command_install_python_modules() {
     # Otherwise, get-pip.py will always try to get latest versions.
     download_get_pip
     echo "# Installing latest pip with preferred setuptools version... #"
-    execute $PYTHON_BIN "$BUILD_DIR"/get-pip.py $PIP_ARGS \
-        pip==$PIP_VERSION --no-setuptools setuptools==$SETUPTOOLS_VERSION
+    execute "$PYTHON_BIN" "$BUILD_DIR"/get-pip.py "${PIP_ARGS[@]}" \
+        pip=="$PIP_VERSION" --no-setuptools setuptools=="$SETUPTOOLS_VERSION"
 
     # pycparser is installed first as setup_requires is ugly.
     # https://pip.pypa.io/en/stable/reference/pip_install/#controlling-setup-requires
     echo "# Installing pycparser with preferred setuptools version... #"
-    execute $PYTHON_BIN -m pip \
-        install $PIP_ARGS -U pycparser==$PYCPARSER_VERSION
+    execute "$PYTHON_BIN" -m pip \
+        install "${PIP_ARGS[@]}" -U pycparser=="$PYCPARSER_VERSION"
 
-    if [ $OS = 'win' ]; then
-        echo "    Skip makefile updating on Windows"
+    if [ "$OS" = "win" ]; then
+        echo -e "\tSkip makefile updating on Windows"
     else
         echo "# Updating Python config Makefile for newly-built Python... #"
-        makefile="$(ls $INSTALL_DIR/lib/$PYTHON_VERSION/config*/Makefile)"
-        makefile_orig="${makefile}.orig"
+        makefile="$(ls "$INSTALL_DIR"/lib/"$PYTHON_VERSION"/config*/Makefile)"
+        makefile_orig="$makefile".orig
 
-        execute cp $makefile $makefile_orig
+        execute cp "$makefile" "$makefile_orig"
         execute sed "s#^prefix=.*#prefix= $INSTALL_DIR#" "$makefile_orig" \
             > "$makefile"
     fi
 
-    for library in $PIP_LIBRARIES ; do
-        execute "$PYTHON_BIN" -m pip install $PIP_ARGS $library
+    for library in "${PIP_LIBRARIES[@]}" ; do
+        execute "$PYTHON_BIN" -m pip install "${PIP_ARGS[@]}" "$library"
     done
 
     # When done, uninstall wheel.
-    execute $PYTHON_BIN -m pip uninstall --yes wheel
+    execute "$PYTHON_BIN" -m pip uninstall --yes wheel
 
     echo "::endgroup::"
 }
 
 
-help_text_test=\
-"Run own tests for the newly-build Python distribution."
+# shellcheck disable=SC2034 # Only used through compgen.
+help_text_test="Run own tests for the newly-build Python distribution."
 command_test() {
     local test_file="test_python_binary_dist.py"
     local python_binary="$PYTHON_BIN"
-    if [ $OS != 'win' ]; then
+
+    if [ ! -d build/ ]; then
+        (>&2 echo "No build/ sub-directory present, try 'build' first!")
+        exit 220
+    fi
+    if [ "$OS" != "win" ]; then
         # Post-cleanup, the binary in /bin is named "python", not "python3.x".
         local python_binary="$INSTALL_DIR/bin/python"
     fi
 
     echo "::group::Chevah tests"
-    echo "#### Executing Chevah tests... ####"
-    execute cp src/chevah-python-test/$test_file "$BUILD_DIR"
-    execute cp src/chevah-python-test/get_binaries_deps.sh "$BUILD_DIR"
+    echo "#### Executing Chevah shell tests... ####"
```

Comment:
Can we have this as a separate script ?

I feel that the current build.sh is one big script that does more than just "build"


We can have a separate script and call it from GitHub Actions only for Linux or Macos.

Reply:
OK with having a separate script. But I also want to execute it locally when testing, not only through GHA.

Reply:
Fixed in today's commits: https://github.com/chevah/pythia/pull/44/files/8ebd0118e9e1c7fb9fa73c52f2c084d93d7bcfc7..d578241af1c86fa3d2f5f38cf93010ed1ace5a5f

The new script can also be executed independently locally.